ext/standard/basic_functions.c

```diff
@@ -1619,71 +1619,71 @@ ZEND_BEGIN_ARG_INFO_EX(arginfo_round, 0, 0, 1)
 	ZEND_ARG_INFO(0, mode)
 ZEND_END_ARG_INFO()
 
-ZEND_BEGIN_ARG_INFO(arginfo_sin, 0)
+ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO(arginfo_sin, IS_DOUBLE, 0)
```

Comment:
All of these will need to be declared nullable due to zpp failures :(

Reply:
@NikiC I think it would be better for the future if we distinguish at the macro level the functions returning nullable types because of ZPP failures and those which may return null even if ZPP failure is changed to `TypeError`. Unless we don't have any of those; I know returning `false` is incredibly common.

Reply:
@nikic @krakjoe bump on this.

Reply:
@morrisonlevi @nikic @krakjoe Is functions return type something that might be happening in `PHP 7.x`? Is so, I've started to [write a script](https://github.com/carusogabriel/php-src/commits/add-return-type) (it is very simple yet), so we can automate that. Let me know if you want me to PR it to core, so we can start add them :smile:

Reply:
I don't think it makes much sense to add return type arginfo, if we're not also adding parameter type arginfo. Furthermore, adding return type arginfo is a general problem as pointed out above, since ZPP failures are handled manually (most functions return NULL in this case, but I've seen functions returning FALSE). It might be most reasonable to postpone this to PHP 8, where we hopefully can throw exceptions on ZPP failures, and maybe even unify arginfo and ZPP.